src/compiler/checker.ts

```diff
@@ -1095,11 +1095,11 @@ module ts {
                     }
                 }
 
-                // Get qualified name 
-                if (enclosingDeclaration &&
-                    // TypeParameters do not need qualification
-                    !(symbol.flags & SymbolFlags.TypeParameter)) {
-
+                // Get qualified name if there is an enclosing declaration and it is not a
+                // type parameter or we have a flag telling us to
+                var needFullyQualifiedName = (enclosingDeclaration && !(symbol.flags & SymbolFlags.TypeParameter));

```

Comment:
Type parameters still shouldn't be qualified even when TypeFormatFlags.UseFullyQualifiedType